app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    caching_result = {}
+
+    @wraps(func)
+    def wrapper(*args) -> dict:
```

Comment:
You should return `Any` for `wrapper` because the result of calculations should not always be dict.